common/src/main/java/org/apache/rocketmq/common/ServiceThread.java

```diff
@@ -133,13 +149,17 @@ protected void waitForRunning(long interval) {
         }
 
         //entry to wait
-        waitPoint.reset();
-
+        lock.lock();
         try {
-            waitPoint.await(interval, TimeUnit.MILLISECONDS);
+            if (!hasNotified.get())
+            {
+                //ignore return value
+                waitPoint.await(interval, TimeUnit.MILLISECONDS);
+            }
         } catch (InterruptedException e) {
             log.error("Interrupted", e);
         } finally {
+            lock.unlock();
```

Comment:
hasNotified.set(false); 
lock.unlock();

if first unlock，may be miss a wait

Reply:
After carefully thinking ,I think you may be right.
fixed.